Merge index_{add,fill,copy,select} sampling

lezcano opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #67833
* #68188
* #67996
* #68523
* #68522
* #69909
* #69908
* #68185
* __->__ #68184
* #68183
* #67014
* #66934
* #67789
* #66933

This was in the TODO list as the three operations are very similar.
Did this as one of them was failing in the noncontig tests and I wanted
to make sure that all of them were tested properly, as they all appear
in the derivative formulas of each other.

After this PR, these operations do pass the noncontiguous tests.

cc @mruberry

Differential Revision: [D32684854](https://our.internmc.facebook.com/intern/diff/D32684854)

lezcano committed:
Merge index_{add,fill,copy,select} sampling

This was in the TODO list as the three operations are very similar.
Did this as one of them was failing in the noncontig tests and I wanted
to make sure that all of them were tested properly, as they all appear
in the derivative formulas of each other.

After this PR, these operations do pass the noncontiguous tests.

[ghstack-poisoned]

lezcano committed:
Update on "Merge index_{add,fill,copy,select} sampling"


This was in the TODO list as the three operations are very similar.
Did this as one of them was failing in the noncontig tests and I wanted
to make sure that all of them were tested properly, as they all appear
in the derivative formulas of each other.

After this PR, these operations do pass the noncontiguous tests.

cc mruberry

[ghstack-poisoned]

lezcano committed:
Update on "Merge index_{add,fill,copy,select} sampling"


This was in the TODO list as the three operations are very similar.
Did this as one of them was failing in the noncontig tests and I wanted
to make sure that all of them were tested properly, as they all appear
in the derivative formulas of each other.

After this PR, these operations do pass the noncontiguous tests.

cc mruberry

[ghstack-poisoned]

lezcano committed:
Update on "Merge index_{add,fill,copy,select} sampling"


This was in the TODO list as the three operations are very similar.
Did this as one of them was failing in the noncontig tests and I wanted
to make sure that all of them were tested properly, as they all appear
in the derivative formulas of each other.

After this PR, these operations do pass the noncontiguous tests.

cc mruberry

[ghstack-poisoned]

lezcano committed:
Update on "Merge index_{add,fill,copy,select} sampling"


This was in the TODO list as the three operations are very similar.
Did this as one of them was failing in the noncontig tests and I wanted
to make sure that all of them were tested properly, as they all appear
in the derivative formulas of each other.

After this PR, these operations do pass the noncontiguous tests.

cc mruberry

[ghstack-poisoned]

lezcano committed:
Update on "Merge index_{add,fill,copy,select} sampling"


This was in the TODO list as the three operations are very similar.
Did this as one of them was failing in the noncontig tests and I wanted
to make sure that all of them were tested properly, as they all appear
in the derivative formulas of each other.

After this PR, these operations do pass the noncontiguous tests.

cc mruberry

[ghstack-poisoned]

mruberry reviewed (approved):
Looks much better

lezcano committed:
Update on "Merge index_{add,fill,copy,select} sampling"


This was in the TODO list as the three operations are very similar.
Did this as one of them was failing in the noncontig tests and I wanted
to make sure that all of them were tested properly, as they all appear
in the derivative formulas of each other.

After this PR, these operations do pass the noncontiguous tests.

cc mruberry

[ghstack-poisoned]

lezcano committed:
Update on "Merge index_{add,fill,copy,select} sampling"


This was in the TODO list as the three operations are very similar.
Did this as one of them was failing in the noncontig tests and I wanted
to make sure that all of them were tested properly, as they all appear
in the derivative formulas of each other.

After this PR, these operations do pass the noncontiguous tests.

cc mruberry

[ghstack-poisoned]

lezcano committed:
Update on "Merge index_{add,fill,copy,select} sampling"


This was in the TODO list as the three operations are very similar.
Did this as one of them was failing in the noncontig tests and I wanted
to make sure that all of them were tested properly, as they all appear
in the derivative formulas of each other.

After this PR, these operations do pass the noncontiguous tests.

cc mruberry

[ghstack-poisoned]

lezcano committed:
Update on "Merge index_{add,fill,copy,select} sampling"


This was in the TODO list as the three operations are very similar.
Did this as one of them was failing in the noncontig tests and I wanted
to make sure that all of them were tested properly, as they all appear
in the derivative formulas of each other.

After this PR, these operations do pass the noncontiguous tests.

cc mruberry

[ghstack-poisoned]

lezcano committed:
Update on "Merge index_{add,fill,copy,select} sampling"


This was in the TODO list as the three operations are very similar.
Did this as one of them was failing in the noncontig tests and I wanted
to make sure that all of them were tested properly, as they all appear
in the derivative formulas of each other.

After this PR, these operations do pass the noncontiguous tests.

cc mruberry

[ghstack-poisoned]

lezcano committed:
Update on "Merge index_{add,fill,copy,select} sampling"


This was in the TODO list as the three operations are very similar.
Did this as one of them was failing in the noncontig tests and I wanted
to make sure that all of them were tested properly, as they all appear
in the derivative formulas of each other.

After this PR, these operations do pass the noncontiguous tests.

cc mruberry

[ghstack-poisoned]

lezcano committed:
Update on "Merge index_{add,fill,copy,select} sampling"


This was in the TODO list as the three operations are very similar.
Did this as one of them was failing in the noncontig tests and I wanted
to make sure that all of them were tested properly, as they all appear
in the derivative formulas of each other.

After this PR, these operations do pass the noncontiguous tests.

cc mruberry

[ghstack-poisoned]

lezcano committed:
Update on "Merge index_{add,fill,copy,select} sampling"


This was in the TODO list as the three operations are very similar.
Did this as one of them was failing in the noncontig tests and I wanted
to make sure that all of them were tested properly, as they all appear
in the derivative formulas of each other.

After this PR, these operations do pass the noncontiguous tests.

cc mruberry

[ghstack-poisoned]

lezcano committed:
Update on "Merge index_{add,fill,copy,select} sampling"


This was in the TODO list as the three operations are very similar.
Did this as one of them was failing in the noncontig tests and I wanted
to make sure that all of them were tested properly, as they all appear
in the derivative formulas of each other.

After this PR, these operations do pass the noncontiguous tests.

cc mruberry

[ghstack-poisoned]

lezcano committed:
Update on "Merge index_{add,fill,copy,select} sampling"


This was in the TODO list as the three operations are very similar.
Did this as one of them was failing in the noncontig tests and I wanted
to make sure that all of them were tested properly, as they all appear
in the derivative formulas of each other.

After this PR, these operations do pass the noncontiguous tests.

cc mruberry

[ghstack-poisoned]

mruberry commented:
@mruberry has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32684854).

lezcano committed:
Update on "Merge index_{add,fill,copy,select} sampling"


This was in the TODO list as the three operations are very similar.
Did this as one of them was failing in the noncontig tests and I wanted
to make sure that all of them were tested properly, as they all appear
in the derivative formulas of each other.

After this PR, these operations do pass the noncontiguous tests.

cc mruberry

Differential Revision: [D32684854](https://our.internmc.facebook.com/intern/diff/D32684854)

[ghstack-poisoned]

lezcano committed:
Update on "Merge index_{add,fill,copy,select} sampling"


This was in the TODO list as the three operations are very similar.
Did this as one of them was failing in the noncontig tests and I wanted
to make sure that all of them were tested properly, as they all appear
in the derivative formulas of each other.

After this PR, these operations do pass the noncontiguous tests.

cc mruberry

Differential Revision: [D32684854](https://our.internmc.facebook.com/intern/diff/D32684854)

[ghstack-poisoned]

lezcano committed:
Update on "Merge index_{add,fill,copy,select} sampling"


This was in the TODO list as the three operations are very similar.
Did this as one of them was failing in the noncontig tests and I wanted
to make sure that all of them were tested properly, as they all appear
in the derivative formulas of each other.

After this PR, these operations do pass the noncontiguous tests.

cc mruberry

Differential Revision: [D32684854](https://our.internmc.facebook.com/intern/diff/D32684854)

[ghstack-poisoned]

lezcano committed:
Update on "Merge index_{add,fill,copy,select} sampling"


This was in the TODO list as the three operations are very similar.
Did this as one of them was failing in the noncontig tests and I wanted
to make sure that all of them were tested properly, as they all appear
in the derivative formulas of each other.

After this PR, these operations do pass the noncontiguous tests.

cc mruberry

Differential Revision: [D32684854](https://our.internmc.facebook.com/intern/diff/D32684854)

[ghstack-poisoned]

lezcano committed:
Update on "Merge index_{add,fill,copy,select} sampling"


This was in the TODO list as the three operations are very similar.
Did this as one of them was failing in the noncontig tests and I wanted
to make sure that all of them were tested properly, as they all appear
in the derivative formulas of each other.

After this PR, these operations do pass the noncontiguous tests.

cc mruberry

Differential Revision: [D32684854](https://our.internmc.facebook.com/intern/diff/D32684854)

[ghstack-poisoned]

lezcano committed:
Update on "Merge index_{add,fill,copy,select} sampling"


This was in the TODO list as the three operations are very similar.
Did this as one of them was failing in the noncontig tests and I wanted
to make sure that all of them were tested properly, as they all appear
in the derivative formulas of each other.

After this PR, these operations do pass the noncontiguous tests.

cc mruberry

Differential Revision: [D32684854](https://our.internmc.facebook.com/intern/diff/D32684854)

[ghstack-poisoned]

lezcano committed:
Update on "Merge index_{add,fill,copy,select} sampling"


This was in the TODO list as the three operations are very similar.
Did this as one of them was failing in the noncontig tests and I wanted
to make sure that all of them were tested properly, as they all appear
in the derivative formulas of each other.

After this PR, these operations do pass the noncontiguous tests.

cc mruberry

Differential Revision: [D32684854](https://our.internmc.facebook.com/intern/diff/D32684854)

[ghstack-poisoned]

lezcano committed:
Update on "Merge index_{add,fill,copy,select} sampling"


This was in the TODO list as the three operations are very similar.
Did this as one of them was failing in the noncontig tests and I wanted
to make sure that all of them were tested properly, as they all appear
in the derivative formulas of each other.

After this PR, these operations do pass the noncontiguous tests.

cc mruberry

Differential Revision: [D32684854](https://our.internmc.facebook.com/intern/diff/D32684854)

[ghstack-poisoned]

lezcano committed:
Update on "Merge index_{add,fill,copy,select} sampling"


This was in the TODO list as the three operations are very similar.
Did this as one of them was failing in the noncontig tests and I wanted
to make sure that all of them were tested properly, as they all appear
in the derivative formulas of each other.

After this PR, these operations do pass the noncontiguous tests.

cc mruberry

Differential Revision: [D32684854](https://our.internmc.facebook.com/intern/diff/D32684854)

[ghstack-poisoned]

lezcano committed:
Update on "Merge index_{add,fill,copy,select} sampling"


This was in the TODO list as the three operations are very similar.
Did this as one of them was failing in the noncontig tests and I wanted
to make sure that all of them were tested properly, as they all appear
in the derivative formulas of each other.

After this PR, these operations do pass the noncontiguous tests.

cc mruberry

Differential Revision: [D32684854](https://our.internmc.facebook.com/intern/diff/D32684854)

[ghstack-poisoned]

lezcano committed:
Update on "Merge index_{add,fill,copy,select} sampling"


This was in the TODO list as the three operations are very similar.
Did this as one of them was failing in the noncontig tests and I wanted
to make sure that all of them were tested properly, as they all appear
in the derivative formulas of each other.

After this PR, these operations do pass the noncontiguous tests.

cc mruberry

Differential Revision: [D32684854](https://our.internmc.facebook.com/intern/diff/D32684854)

[ghstack-poisoned]

mruberry commented:
@mruberry has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32684854).

lezcano committed:
Update on "Merge index_{add,fill,copy,select} sampling"


This was in the TODO list as the three operations are very similar.
Did this as one of them was failing in the noncontig tests and I wanted
to make sure that all of them were tested properly, as they all appear
in the derivative formulas of each other.

After this PR, these operations do pass the noncontiguous tests.

cc mruberry

Differential Revision: [D32684854](https://our.internmc.facebook.com/intern/diff/D32684854)

[ghstack-poisoned]

mruberry commented:
@mruberry has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32684854).

lezcano committed:
Update on "Merge index_{add,fill,copy,select} sampling"


This was in the TODO list as the three operations are very similar.
Did this as one of them was failing in the noncontig tests and I wanted
to make sure that all of them were tested properly, as they all appear
in the derivative formulas of each other.

After this PR, these operations do pass the noncontiguous tests.

cc mruberry

Differential Revision: [D32684854](https://our.internmc.facebook.com/intern/diff/D32684854)

[ghstack-poisoned]

mruberry commented:
@mruberry has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32684854).

lezcano committed:
Update on "Merge index_{add,fill,copy,select} sampling"


This was in the TODO list as the three operations are very similar.
Did this as one of them was failing in the noncontig tests and I wanted
to make sure that all of them were tested properly, as they all appear
in the derivative formulas of each other.

After this PR, these operations do pass the noncontiguous tests.

cc mruberry

Differential Revision: [D32684854](https://our.internmc.facebook.com/intern/diff/D32684854)

[ghstack-poisoned]

mruberry commented:
@mruberry has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32684854).

lezcano committed:
Update on "Merge index_{add,fill,copy,select} sampling"


This was in the TODO list as the three operations are very similar.
Did this as one of them was failing in the noncontig tests and I wanted
to make sure that all of them were tested properly, as they all appear
in the derivative formulas of each other.

After this PR, these operations do pass the noncontiguous tests.

cc mruberry

Differential Revision: [D32684854](https://our.internmc.facebook.com/intern/diff/D32684854)

[ghstack-poisoned]

lezcano committed:
Update on "Merge index_{add,fill,copy,select} sampling"


This was in the TODO list as the three operations are very similar.
Did this as one of them was failing in the noncontig tests and I wanted
to make sure that all of them were tested properly, as they all appear
in the derivative formulas of each other.

After this PR, these operations do pass the noncontiguous tests.

cc mruberry

Differential Revision: [D32684854](https://our.internmc.facebook.com/intern/diff/D32684854)

[ghstack-poisoned]

lezcano committed:
Update on "Merge index_{add,fill,copy,select} sampling"


This was in the TODO list as the three operations are very similar.
Did this as one of them was failing in the noncontig tests and I wanted
to make sure that all of them were tested properly, as they all appear
in the derivative formulas of each other.

After this PR, these operations do pass the noncontiguous tests.

cc mruberry

Differential Revision: [D32684854](https://our.internmc.facebook.com/intern/diff/D32684854)

[ghstack-poisoned]

lezcano committed:
Update on "Merge index_{add,fill,copy,select} sampling"


This was in the TODO list as the three operations are very similar.
Did this as one of them was failing in the noncontig tests and I wanted
to make sure that all of them were tested properly, as they all appear
in the derivative formulas of each other.

After this PR, these operations do pass the noncontiguous tests.

cc mruberry

Differential Revision: [D32684854](https://our.internmc.facebook.com/intern/diff/D32684854)

[ghstack-poisoned]

lezcano committed:
Update on "Merge index_{add,fill,copy,select} sampling"


This was in the TODO list as the three operations are very similar.
Did this as one of them was failing in the noncontig tests and I wanted
to make sure that all of them were tested properly, as they all appear
in the derivative formulas of each other.

After this PR, these operations do pass the noncontiguous tests.

cc mruberry

Differential Revision: [D32684854](https://our.internmc.facebook.com/intern/diff/D32684854)

[ghstack-poisoned]

mruberry commented:
@mruberry has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32684854).

lezcano committed:
Update on "Merge index_{add,fill,copy,select} sampling"


This was in the TODO list as the three operations are very similar.
Did this as one of them was failing in the noncontig tests and I wanted
to make sure that all of them were tested properly, as they all appear
in the derivative formulas of each other.

After this PR, these operations do pass the noncontiguous tests.

cc mruberry

Differential Revision: [D32684854](https://our.internmc.facebook.com/intern/diff/D32684854)

[ghstack-poisoned]

mruberry commented:
@mruberry has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32684854).

lezcano committed:
Update on "Merge index_{add,fill,copy,select} sampling"


This was in the TODO list as the three operations are very similar.
Did this as one of them was failing in the noncontig tests and I wanted
to make sure that all of them were tested properly, as they all appear
in the derivative formulas of each other.

After this PR, these operations do pass the noncontiguous tests.

cc mruberry

Differential Revision: [D32684854](https://our.internmc.facebook.com/intern/diff/D32684854)

[ghstack-poisoned]

lezcano committed:
Update on "Merge index_{add,fill,copy,select} sampling"


This was in the TODO list as the three operations are very similar.
Did this as one of them was failing in the noncontig tests and I wanted
to make sure that all of them were tested properly, as they all appear
in the derivative formulas of each other.

After this PR, these operations do pass the noncontiguous tests.

cc mruberry

Differential Revision: [D32684854](https://our.internmc.facebook.com/intern/diff/D32684854)

[ghstack-poisoned]

lezcano committed:
Update on "Merge index_{add,fill,copy,select} sampling"


This was in the TODO list as the three operations are very similar.
Did this as one of them was failing in the noncontig tests and I wanted
to make sure that all of them were tested properly, as they all appear
in the derivative formulas of each other.

After this PR, these operations do pass the noncontiguous tests.

cc mruberry

Differential Revision: [D32684854](https://our.internmc.facebook.com/intern/diff/D32684854)

[ghstack-poisoned]

lezcano committed:
Update on "Merge index_{add,fill,copy,select} sampling"


This was in the TODO list as the three operations are very similar.
Did this as one of them was failing in the noncontig tests and I wanted
to make sure that all of them were tested properly, as they all appear
in the derivative formulas of each other.

After this PR, these operations do pass the noncontiguous tests.

cc mruberry

Differential Revision: [D32684854](https://our.internmc.facebook.com/intern/diff/D32684854)

[ghstack-poisoned]

mruberry commented:
@mruberry has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32684854).

lezcano committed:
Update on "Merge index_{add,fill,copy,select} sampling"


This was in the TODO list as the three operations are very similar.
Did this as one of them was failing in the noncontig tests and I wanted
to make sure that all of them were tested properly, as they all appear
in the derivative formulas of each other.

After this PR, these operations do pass the noncontiguous tests.

cc mruberry

Differential Revision: [D32684854](https://our.internmc.facebook.com/intern/diff/D32684854)

[ghstack-poisoned]

mruberry commented:
@mruberry has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32684854).

lezcano committed:
Update on "Merge index_{add,fill,copy,select} sampling"


This was in the TODO list as the three operations are very similar.
Did this as one of them was failing in the noncontig tests and I wanted
to make sure that all of them were tested properly, as they all appear
in the derivative formulas of each other.

After this PR, these operations do pass the noncontiguous tests.

cc mruberry

Differential Revision: [D32684854](https://our.internmc.facebook.com/intern/diff/D32684854)

[ghstack-poisoned]

mruberry commented:
@mruberry has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32684854).

lezcano committed:
Update on "Merge index_{add,fill,copy,select} sampling"


This was in the TODO list as the three operations are very similar.
Did this as one of them was failing in the noncontig tests and I wanted
to make sure that all of them were tested properly, as they all appear
in the derivative formulas of each other.

After this PR, these operations do pass the noncontiguous tests.

cc mruberry

Differential Revision: [D32684854](https://our.internmc.facebook.com/intern/diff/D32684854)

[ghstack-poisoned]

mruberry commented:
@mruberry has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32684854).

lezcano committed:
Update on "Merge index_{add,fill,copy,select} sampling"


This was in the TODO list as the three operations are very similar.
Did this as one of them was failing in the noncontig tests and I wanted
to make sure that all of them were tested properly, as they all appear
in the derivative formulas of each other.

After this PR, these operations do pass the noncontiguous tests.

cc mruberry

Differential Revision: [D32684854](https://our.internmc.facebook.com/intern/diff/D32684854)

[ghstack-poisoned]

mruberry commented:
@mruberry has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32684854).
